### PR TITLE
automatically add labels to pull requests based on changed files

### PR DESCRIPTION
**Pull request by zacharyburnett**
This GitHub workflow runs a labeler action on new / edited pull requests that scans the filenames of changed files and applies relevant labels. This process is configured in `.github/labeler.yml`, which I copied and modified from https://github.com/spacetelescope/stsynphot_refactor/blob/master/.github/labeler.yml

An example of this labeler working can be seen here: https://github.com/spacetelescope/stsynphot_refactor/pull/153
image

**Checklist**
- [ ] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [ ] added relevant milestone
- [x] added relevant label(s)

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  - 'jwst/**/*wfss*/**/*'
+
+NIRSPEC:
+  - 'jwst/**/*nirspec*/**/*'
+
+NIRCAM:
+  - 'jwst/**/*nircam*/**/*'
+
+NIRISS:
+  - 'jwst/**/*niriss*/**/*'
```
Comment (zacharyburnett) [inline .github/labeler.yml:211]: I'm not sure if each instruments has specific files that are exclusive to them, so I used general patterns here.

Comment (hbushouse) [reply .github/labeler.yml]: It's rare, but there are a few instances here and there of modules that have an instrument-specific name.

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+barshadow:
+  - 'jwst/barshadow/**'
+
+combine_1d:
+  - 'jwst/combine_1d/**'
+
+coron3 pipeline:
+  - 'jwst/coron/**'
+  - jwst/pipeline/calwebb_coron.py
```
Comment (hbushouse) [inline .github/labeler.yml:37]: I think the actual module name is calwebb_coron**3**.py

File: .github/labeler.yml [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+mrs_imatch:
+  - 'jwst/mrs_imatch/**'
+
+msa_flagging:
+  - 'jwst/msaflagopen/**'
+
+outlier_detection:
+  - 'jwst/outlier_detection/**'
+
+path loss:
```
Comment (hbushouse) [inline .github/labeler.yml:114]: does the pathloss label have a whitespace in the middle of it?

File: .github/labeler.yml [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+  - 'jwst/**/*nirspec*'
+  - 'jwst/**/*nirspec*/**'
+
+NIRCAM:
+  - 'jwst/**/*nircam*'
+  - 'jwst/**/*nircam*/**'
+
+NIRISS:
+  - 'jwst/**/*niriss*'
+  - 'jwst/**/*niriss*/**'
```
Comment (hbushouse) [inline .github/labeler.yml:215]: You're missing FGS and MIRI.

**Review by hbushouse — COMMENTED** [review]
Looks OK, other than minor comments/questions?  Will be eager to see how it actually works.

File: .github/labeler.yml (hunk 3 above)
Comment (zacharyburnett) [reply .github/labeler.yml:114]: yes indeed it is `path loss`, you can see here: https://github.com/spacetelescope/jwst/labels?q=path

Should the label be renamed?

Comment (hbushouse) [reply .github/labeler.yml:114]: No, that's OK. I just didn't realize it was spelled that way.

**Comment by zacharyburnett** [comment]
> Looks OK, other than minor comments/questions? Will be eager to see how it actually works.

Yeah, it should make PRs easier!

File: .github/labeler.yml [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+assign_wcs:
+  - 'jwst/assign_wcs/**'
+
+associations:
+  - 'jwst/associations/**'
+
+background:
+  - 'jwst/background/**'
+
+barshadow:
```
Comment (hbushouse) [inline .github/labeler.yml:29]: It looks like the "bar shadow" label also has a whitespace in it.

File: .github/labeler.yml [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+  - 'jwst/barshadow/**'
+
+combine_1d:
+  - 'jwst/combine_1d/**'
+
+coron3 pipeline:
```
Comment (hbushouse) [inline .github/labeler.yml:35]: could add similar entries for the other pipelines: image2, image3, spec2, spec3, ami3, tso3, detector1

**Review by nden — APPROVED** [review]

**Pull request merged**